### ISSUE TITLE
Problem with message:  No template found for residue 44 (NME).  The set of atoms matches NME, but the bonds are different.

### ISSUE THREAD

**Issue by BillSwope** [openmm/openmm#3370]
I have a protein with an NME termination, and it appears the OpenMM code can "almost" but not quite recognize it.  
I am using amber14-all.xml.  Here is  the message from forcefield.py
raise ValueError('No template found for residue %d (%s).  %s' % (res.index+1, res.name, _findMatchErrors(self, res)))
ValueError: No template found for residue 44 (NME).  The set of atoms matches NME, but the bonds are different.

The CONECT list looks ok, so I don't see why it is not finding  the NME.
Here are the relevent lines from the pdb file:
.....
ATOM    665  C   CYS A 258     113.948 -25.346 103.859  1.00 58.44           C
ATOM    666  O   CYS A 258     114.838 -26.125 104.200  1.00 45.89           O
ATOM    667  CB  CYS A 258     111.898 -26.805 104.121  1.00 70.56           C
ATOM    668  SG  CYS A 258     110.239 -26.926 104.857  1.00 85.29           S
ATOM    669  H   CYS A 258     111.777 -24.341 102.776  1.00 49.31           H
ATOM    670  HA  CYS A 258     112.603 -25.299 105.501  1.00 59.42           H
ATOM    671  HB2 CYS A 258     111.816 -26.983 103.048  1.00 70.56           H
ATOM    672  HB3 CYS A 258     112.501 -27.615 104.535  1.00 70.56           H
ATOM    673  HG  CYS A 258     109.670 -26.003 104.077  1.00 85.29           H
ATOM    674  N   NME A 259     114.190 -24.383 102.970  1.00  0.00           N
ATOM    675  CH3 NME A 259     115.491 -24.198 102.362  1.00  0.00           C
ATOM    676  H   NME A 259     113.438 -23.760 102.711  1.00  0.00           H
ATOM    677 HH31 NME A 259     115.674 -23.134 102.209  1.00  0.00           H
ATOM    678 HH32 NME A 259     116.260 -24.608 103.017  1.00  0.00           H
ATOM    679 HH33 NME A 259     115.520 -24.713 101.402  1.00  0.00           H
ATOM    680  N   PRO A 264     125.025 -27.914 101.651  1.00 50.45           N1+
ATOM    681  CA  PRO A 264     125.642 -27.938 100.326  1.00 58.62           C
ATOM    682  C   PRO A 264     125.656 -26.510  99.707  1.00 62.18           C
ATOM    683  O   PRO A 264     124.623 -26.063  99.200  1.00 54.41           O
...
ATOM   3437 HH21 ARG A 446     135.440  -6.389  92.662  1.00 64.82           H
ATOM   3438 HH22 ARG A 446     134.745  -4.928  91.993  1.00 64.82           H
TER    3439      ARG A 446
CONECT  674  675  676
CONECT  675  674  677  678  679
CONECT  676  674
CONECT  677  675
CONECT  678  675
CONECT  679  675
CONECT 2311 2312 2317
CONECT 2312 2311 2313 2315 2318
...

Any suggestions are welcome.

Regards....Bill

**Comment by peastman** [comment]
I think you want to remove the CONECT records for it.  NME is considered a standard residue, so bonds are inferred automatically.

**Comment by BillSwope** [comment]
Thanks, Peter
I'll give it a try and report back....Bill

**Comment by BillSwope** [comment]
Peter, that didn't work.  I got exactly the same message.  This is
surprising, since "The set of atoms matches NME, but the bonds are
different." when there are not CONECT fields in the PDB seems to imply the
bond list constructed from inter-site distances might be messed up.

In case it matters, I produced the pdb file with MOE.  I edited a pdb file
to change a terminal CYS residue that was "aldehyde terminated", and which
was throwing similar OpenMM errors, to a chain that was terminated with
---CYS-NME.

Thanks a lot for the help, Peter.....
Bill

**Comment by peastman** [comment]
Could you post the PDB file?  I'll take a look at it.

**Comment by BillSwope** [comment]
I will ask.  The protein pdb file is of a receptor that is part of an
active drug program, so they might not want to let it out.  If it is close
to one from the actual PDB and I can reproduce the error, I can try to get
you that.

Is there any test you'd like me to try in the meantime?

Regards....Bill

**Comment by peastman** [comment]
Looking more closely at the part of the file you quoted, I noticed something strange.  NME appears in the middle of a chain???  It's supposed to be a terminal group.  Did you intend there to be a break between chains?  If so, you can indicate it by adding a TER record and having the chain ID change.

**Comment by BillSwope** [comment]
Good points, Peter. 

I got this case from someone trying to use some openmm code I wrote. I’m finding that almost everyone here at Genentech has their own set of modeling tools (various combinations of Schrodinger, MOE, OpenEye, others, etc) and, with all the different targets across therapeutic teams, there’s a lot of complexity going on. I’ve seen that MOE and Maestro supports multiple chains with the same chainID.  I’ll ask around and see if there’s a reason for this. And I’ll try changing the ChainID and adding a TER record and see if that fixes any of these issues.  
And thanks again for the suggestions. 
…Bill

**Comment by peastman** [comment]
OpenMM tries to be flexible about how it reads PDB files.  According to the spec you should both have a TER record and change the ID.  But if you do just one or the other, that will get interpreted as a new chain.

**Comment by BillSwope** [comment]
I’m wondering if my colleague is docking into a protein structure with gaps from flexible loops that were poorly characterized in the X-ray structure. That might explain the gaps and not incrementing the ChainID. I’m surprised this sort of thing doesn’t throw all kinds of complaints from the other tools. I put it into MOE and everything was just peachy.

**Comment by BillSwope** [comment]
Peter, you'll love this.

Yes, there are breaks in the chain.  We're looking into this.  However,
changing the name from NMA (for N-Methyl-Acetamide) to NME (for N-MEthyl
acetamide) seems to have helped.  Apparently, Schrodinger tools use NMA,
but AMBER tools use NME.  I stared at this for a long time and never
noticed it.

Once I verify this, I will update my post in the GitHub.

Regards....Bill

**Comment by BillSwope** [comment]
And, of course, there is more.  The problem turned out not to be the breaks
in the chain, nor the NMA vs NME issue.  The protein had a covalently
bonded ligand to the sulfur of a CYS.  And the editing of that out of the
X-ray structure caused the problems.  It is pretty specific to this
particular protein case.

Bill

**Comment by peastman** [comment]
PDB files are so much fun!  For what it's worth, NME is the standard name in the chemical components dictionary.  NMA is something totally different.  I would send a bug report to Schrodinger telling them they're using the wrong name.

**Comment by peastman** [comment]
I'm closing this since I think everything has been answered.  Feel free to reopen it if I missed something!

**Comment by BillSwope** [comment]
Thanks, Peter.

BTW, I just started using SystemGenerator, which John says is relatively new. Goal is to simulate with Openmm from a PDBs prepared using Maestro or OpenEye tools or MOE tools. So far, working well on MOE systems. I see if you have more than 9999 water molecules (each treated as a separate “residue”) the four character field in the PDB is interpreted as a hexidecimal number. This is not what MOE does. It switches to a base-36 number using digits 0 to 9 and all the uppercase letters A to Z. It also wasn’t letting each ion be a separate residue, which I think it should. So I wrote a cleanup code that fixes these things and with the fixed PDB file openmm seems to work. 

Regards…Bill
